File: card-core/src/main/java/com/jean202/cardmizer/core/port/in/GetSpendingRecordsUseCase.java
package com.jean202.cardmizer.core.port.in;

import com.jean202.cardmizer.core.domain.CardId;
import com.jean202.cardmizer.core.domain.SpendingPeriod;
import com.jean202.cardmizer.core.domain.SpendingRecord;
import java.util.List;

public interface GetSpendingRecordsUseCase {
    List<SpendingRecord> getByPeriod(SpendingPeriod period, CardId cardId);
}

File: card-core/src/main/kotlin/com/jean202/cardmizer/core/port/in/GetSpendingRecordsUseCase.kt
package com.jean202.cardmizer.core.port.`in`

import com.jean202.cardmizer.core.domain.CardId
import com.jean202.cardmizer.core.domain.SpendingPeriod
import com.jean202.cardmizer.core.domain.SpendingRecord

fun interface GetSpendingRecordsUseCase {
    fun getByPeriod(period: SpendingPeriod, cardId: CardId?): List<SpendingRecord>
}
